h/settings.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+
+"""Helpers for parsing settings from the environment."""
+
+import string
+
+
+class SettingError(Exception):
+    pass
+
+
+class EnvSetting(object):
+
+    """An (optionally typed) setting read from an environment variable."""
+
+    def __init__(self, setting, varname, type=None):
+        self.setting = setting
+        self.varname = varname
+        if type is not None:
+            self.type = type
+        else:
+            self.type = str
+
+    def __call__(self, environ):
+        if self.varname in environ:
+            try:
+                value = self.type(environ[self.varname])
+            except ValueError:
+                raise SettingError('error parsing environment variable '
+                                   '{varname}={value!r} as {typename}'.format(
+                                       varname=self.varname,
+                                       typename=self.type.__name__,
+                                       value=environ[self.varname]))
+            return {self.setting: value}
+
+
+class DockerSetting(object):
+
+    """A setting read from Docker link environment variables."""
+
+    def __init__(self, setting, link, pattern):
+        self.setting = setting
+        self.link = link.upper()
+        self.pattern = pattern
+
+        # Determine the settings that need to be present
+        formatter = string.Formatter()
+        self.placeholders = [x for _, x, _, _ in formatter.parse(pattern)
```

Comment:
I think it would be clearer to name this `field_name` or something along those lines rather than `x` for clarity.